app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,79 @@
+
+
+class TripsController < ApplicationController
+  def show
+    @trip = Trip.find_by(id: params[:id])
+    if @trip.nil? 
+    # unless !@trip.nil?
+      redirect_to trips_path
+    end
+  end
+
+  def new
+    @trip = Trip.new
+  end
+
+  def create
+    passenger = Passenger.find_by(id: params[:trip][:passenger_id])
```

Comment:
passenger_id is a route parameter, **not** a field submitted by a form.  Instead this should be:

`    passenger = Passenger.find_by(id: params[:passenger_id])`